### PR TITLE
feat: Disable js toggle button if field is not empty

### LINKED ISSUE #10833: [Feature]-[9884]:Disable JS button if the field is not empty or if the default text is modified
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Disable JS button in property pane controls if the field is not empty or if the default text has changed.
Also, add a popup on the disabled JS button to indicate why it is disabled.

### Why should this be worked on?

The JS toggle interaction is extremely destructive. The JS binding is getting cleared if press the JS mode toggle button. Please see attached screen recording:

https://user-images.githubusercontent.com/10436935/152168497-22754eaf-c1ab-43e1-a010-602b4ca7fc93.mov

See the Slack discussions [here](https://theappsmith.slack.com/archives/C025SE88KNE/p1643799250836589)

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Disable the JS toggle button if the value is non-empty and non-default value.
This is done to avoid the loss of the user's code on accidentally clicking the JS toggle button.

Fixes #10833 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/disable-js-toggle 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.49 **(0)** | 37.98 **(-0.01)** | 36.28 **(0.01)** | 56.7 **(-0.01)**
 :green_circle: | app/client/src/components/ads/index.ts | 100 **(0)** | 100 **(0)** | 7.41 **(1.85)** | 100 **(0)**
 :green_circle: | app/client/src/components/propertyControls/IconSelectControl.tsx | 91.37 **(0.07)** | 76.11 **(0)** | 95.83 **(0)** | 91.41 **(0.07)**
 :red_circle: | app/client/src/components/propertyControls/StyledControls.tsx | 68 **(0.88)** | 69.74 **(-6.02)** | 45.45 **(1.26)** | 68 **(0.88)**
 :red_circle: | app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx | 50.32 **(-1.59)** | 32.32 **(-2.56)** | 20 **(-1.43)** | 51.01 **(-1.84)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx [hunk 1]
```diff
@@ -475,6 +476,24 @@ const PropertyControl = memo((props: Props) => {
     };
 
     const uniqId = btoa(`${widgetProperties.widgetId}.${propertyName}`);
+    const canDisplayValueInUI = PropertyControlFactory.controlUIToggleValidation.get(
+      config.controlType,
+    );
+
+    let isToggleDisabled = false;
+    if (
```
Comment (rahulramesha) [inline app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx:484]: Could you please add some comments here and the condition below

File: app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx [hunk 2]
```diff
@@ -496,15 +515,24 @@ const PropertyControl = memo((props: Props) => {
...
             />
             {isConvertible && (
-              <JSToggleButton
-                active={isDynamic}
-                className={`focus:ring-2 t--js-toggle ${
-                  isDynamic ? "is-active" : ""
-                }`}
-                onClick={() => toggleDynamicProperty(propertyName, isDynamic)}
+              <Tooltip
+                content="A dynamic or invalid value cannot be represented in the UI mode. Please update the value or clear the value to toggle back to UI mode."
```
Comment (rahulramesha) [inline app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx:519]: Could you use `messages.ts` to store the text and access it here.

**Review by rahulramesha — CHANGES_REQUESTED** [review]

File: app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx:485]: done. Can you check?

**Review by rahulramesha — DISMISSED** [review]

**Comment by Sripriya93** [comment]
Checkbox group Widget "Options" JS toggle is affected by this feature.

https://images.zenhubusercontent.com/275993885/ffd69a3b-843f-41d2-8a8b-96ade483f6b6/js_toggle___editor___appsmith___9_june_2022.mp4

**Review by rahulramesha — APPROVED** [review]

**Pull request merged**